src/docs/tool-libfuzzer.dox

```diff
@@ -0,0 +1,210 @@
+/*! @page tool-libfuzzer Testing with LLVM LibFuzzer
+
+# Building and running an LLVM LibFuzzer target
+
+LLVM LibFuzzer is in-process, coverage-guided, evolutionary fuzzing engine. It feeds a series of
+fuzzed inputs via a "target" function and attempts to trigger crashes, memory bugs and undefined
+behavior.
+
+## Step 1: Select Clang as your C compiler in your build configuration
+
+Support for LibFuzzer is implemented as a compiler flag in Clang. Our configuration checks whether
+the compiler in use supports \c -fsanitize=fuzzer and if so, elects to build the fuzz targets.
+
+Compiling with the ASan isn't mandatory but it is recommended since fuzzing often exposes memory
+bugs.
+
+@code
+$ cd build_posix/
+$ ../configure CC="clang-8" CFLAGS="-fsanitize=address"
+@endcode
```

Comment:
Yes, I'd say we do want that. We supply `-fsanitize=fuzzer` behind the scenes because it's required to actually build the fuzz target (Clang provides its own `main` in this case) so that is quite different.

ASan is a sweet spot of functionality vs speed but you might equally want to run MSan or UBSan. I don't think the build system should be enforcing that you build with a particular sanitizer.